apollo-compiler/src/main/kotlin/com/apollographql/apollo3/compiler/codegen/java/model/ModelBuilder.kt

```diff
@@ -103,4 +112,42 @@ internal class ModelBuilder(
       }
     }
   }
+
+  private fun TypeSpec.addBuilder(context: JavaContext): TypeSpec {
+    val fields = fieldSpecs.filter { !it.modifiers.contains(Modifier.STATIC) }
+      .filterNot { it.name.startsWith(prefix = "$") }
+    if (fields.isEmpty()) {
+      return this
+    } else {
+      val builderVariable = JavaClassNames.Builder.simpleName().decapitalizeFirstLetter()
+      val builderClass = ClassName.get("", JavaClassNames.Builder.simpleName())
+      val toBuilderMethod = MethodSpec.methodBuilder(Builder.TO_BUILDER_METHOD_NAME)
+        .addModifiers(Modifier.PUBLIC)
+        .returns(builderClass)
+        .addStatement("\$T \$L = new \$T()", builderClass, builderVariable, builderClass)
+        .addCode(fields
+          .map { CodeBlock.of("\$L.\$L = \$L;\n", builderVariable, it.name, it.type.unwrapOptionalValue(it.name)) }
```

Comment:
Is there a specific reason you call `unwrapOptionalValue` here? Looks like it's only required for `Optional<T>` values which are not generated right now (see also https://github.com/apollographql/apollo-kotlin/issues/4337)

Even if `Optional` were generated, I think it's fine to have the builder field have the same type as the model? It's more consistent? 


```java
  public static class Animal {
    public Optional<String> name;
  }

  public static final class Builder {
    private Optional<String> name;
  }

```

Reply:
I guess it boils down to whether we want to be able to pass null to a previously non-null builder:

```java
Data.Builder builder = new Data.Builder();

builder.name(Optional.of("Luke"));

// ... some code

// If we want to be able to do this ultimately, we'll need to keep the  `Optional` wrapper
builder.name(Optional.ofNullable(null))

```

Reply:
TBH I just copied this code from Apollo Android without trying to understand it too much. 😅 I'll remove it and see if it has any effect.